pkg/processor/schemaregistry/avro/schema.go

```diff
@@ -0,0 +1,99 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package avro
+
+import (
+	"reflect"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/hamba/avro/v2"
+	"github.com/lovromazgon/franz-go/pkg/sr"
+)
+
+const Type = sr.TypeAvro
+
+// Schema represents an Avro schema. It exposes methods for marshaling and
+// unmarshaling data.
+type Schema struct {
+	schema        avro.Schema
+	unionResolver UnionResolver
```

Comment:
I added a comment 🙂 It's because of how unions are marshled / unmarshaled in the lib we're using (unfortunately other libs work the same, otherwise I would have switched).